File: prolog/p-36.pl
:- ensure_loaded('p-09').
:- ensure_loaded('p-35').

change([X|Xs], Res) :-
    length([X|Xs], L), Res = X-L.

prime_factors_mult(N, Ans) :-
    prime_factors(N, Xs),
    pack(Xs, Ys),
    maplist(change, Ys, Ans).

File: erlang/p_36.erl
-module(p_36).
-export([prime_factors_mult/1]).
-import(p_35, [prime_factors/1]).
-import(p_09, [pack/1]).

prime_factors_mult(N) ->
    Xs = prime_factors(N),
    Ys = pack(Xs),
    lists:map(fun ([Z|Zs]) -> {Z, length(Zs) + 1} end, Ys).
